lib/new_relic/agent/instrumentation/controller_instrumentation.rb

```diff
@@ -165,10 +165,6 @@ def newrelic_read_attr(attr_name) # :nodoc:
           def add_transaction_tracer(method, options = {})
             NewRelic::Agent.record_api_supportability_metric(:add_transaction_tracer)
 
```

Comment:
We were doing work to set these variables before doing a conditional check that could potentially return early. These variables are not involved in the check, so it is a waste to set them before the check has completed.